databuilder/docs/specs.py

```diff
@@ -119,6 +118,52 @@ def build_paragraph(paragraph_id, test_fn):
     )
 
 
+def get_series_code(source_lines, series_index):
+    """
+    Extract the definition of the series from the test function.
+    """
+    # A series may be defined over more than one line; iterate over the next
+    # lines and build the string representing the series definition; when a potential
+    # ending (a , at the end of a line) is found, check that the parentheses in the
+    # statement so far are balanced; if they are not, we haven't reached the end of the
+    # definition yet.
+    first_series_line = source_lines[series_index]
+    # Find the leading whitespace for the first line; this will be stripped, but in order to
+    # preserve indentation, we strip only the equivalent whitespace from any subsequent lines.
+    leading_whitespace_match = re.match(r"^(?P<whitespace>\s+)\w*", first_series_line)
+    if leading_whitespace_match:
+        leading_whitespace = leading_whitespace_match.group("whitespace")
+    else:
+        leading_whitespace = ""
+    series_lines = []
+    for line in source_lines[series_index:]:
+        series_lines.append(line.rstrip().replace(leading_whitespace, ""))
+        series_line = "".join(series_lines)
+        if series_line.strip().endswith(","):
+            # check series_line is balanced; if it is, then we're done
+            if series_is_balanced(series_line):
+                break
+    return "\n".join(series_lines).strip().removesuffix(",")
+
+
+def series_is_balanced(series_line_string):
+    """
+    Takes a string representing a series definition and ensures that parentheses are
+    balanced.
+    """
+    stack = []
+    for i in series_line_string:
```

Comment:
To me, `i` is a surprising variable name -- it suggests an integer value.  Maybe `c` or `char`?